Error [ERR_MODULE_NOT_FOUND] in useUppyState.js in Vite Remix Application

### Initial checklist

- [X] I understand this is a bug report and questions should be posted in the [Community Forum](https://community.transloadit.com/)
- [X] I searched [issues](https://github.com/transloadit/uppy/issues?q=is%3Aissue) and couldn’t find anything (or linked relevant results below)

### Link to runnable example

Uppy Dashboard

### Steps to reproduce

Use a basic Vite Based Remix Application or any Vite Based React Application. 

Install @Uppy/Dashboard

Run locally and make sure everything is working fine. 

Now create a server PROD build and run the PROD build you get the following error:

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/myapp/node_modules/use-sync-external-store/with-selector' imported from /myapp/node_modules/@uppy/react/lib/useUppyState.js

### Expected behavior

There should not be any errors and the application should run in PROD as in DEV server as well.

### Actual behavior

Server run fails with the following error:

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/myapp/node_modules/use-sync-external-store/with-selector' imported from /myapp/node_modules/@uppy/react/lib/useUppyState.js

--- a/packages/@uppy/react/src/useUppyState.ts
+++ b/packages/@uppy/react/src/useUppyState.ts
@@ -1,7 +1,7 @@
 import type { Uppy, State } from '@uppy/core'
 import type { Body, Meta } from '@uppy/utils/lib/UppyFile'
 import { useMemo, useCallback } from 'react'
-import { useSyncExternalStoreWithSelector } from 'use-sync-external-store/with-selector'
+import { useSyncExternalStoreWithSelector } from 'use-sync-external-store/with-selector.js'
 
 export default function useUppyState<
   M extends Meta = Meta,

